datafusion/core/tests/sql/group_by.rs

```diff
@@ -68,6 +69,173 @@ async fn group_by_date_trunc() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn group_by_limit() -> Result<()> {
```

Comment:
I wonder if we can implement most of these tests in `sqllogictest`?

Reply:
I added some `sqllogictest`s. 

I think we should keep this test though, because it asserts the `lim=[X]` is present in the physical plan, which is the only way we can know for (almost?) sure our optimized stream type is actually running. Its results should be indistinguishable from the unoptimized version except for memory usage.